concurrency-limits-core/src/main/java/com/netflix/concurrency/limits/limiter/AbstractPartitionedLimiter.java

```diff
@@ -216,6 +221,7 @@ public Optional<Listener> acquire(ContextT context) {
         try {
             lock.lock();
             if (shouldBypass(context)){
+                lock.unlock();
```

Comment:
Do we even need to lock in the `shouldBypass` scenario?

Reply:
The issue I was guarding for is that it's not enforced. That [could be overriden by any implementation](https://github.com/Netflix/concurrency-limits/blob/075bc76be8b90172150699dbb2f28c6f85c71e97/concurrency-limits-core/src/main/java/com/netflix/concurrency/limits/limiter/AbstractLimiter.java#L116-L118).

I think if we got shouldBypass removed from the lock, then the only reason we need the lock is for getInflight >= getLimit check, and I _think_ we can do that lock-free as well.

Reply:
https://github.com/Netflix/concurrency-limits/blob/075bc76be8b90172150699dbb2f28c6f85c71e97/concurrency-limits-core/src/main/java/com/netflix/concurrency/limits/limiter/AbstractLimiter.java#L73-L74

Would it be a viable solution to expand the scope of this comment to indicate that the predicate/subclasses should not query the limiter, and if they do so they are responsible for concurrency?

About the only situation I can see this happening is where the shouldBypass might make a dynamic decision based on the `getInflight` value. I'm not sure what real-world configurations exist.

Reply:
Looking over at these examples -- [GrpcServerLimiterBuilder.java](https://github.com/Netflix/concurrency-limits/blob/075bc76be8b90172150699dbb2f28c6f85c71e97/concurrency-limits-grpc/src/main/java/com/netflix/concurrency/limits/grpc/server/GrpcServerLimiterBuilder.java#L49-L93) and [ServletLimiterBuilder.java](https://github.com/Netflix/concurrency-limits/blob/075bc76be8b90172150699dbb2f28c6f85c71e97/concurrency-limits-servlet/src/main/java/com/netflix/concurrency/limits/servlet/ServletLimiterBuilder.java#L77-L143) it seems that none of them are built to refer back to the limiter itself. If this is the expected & existing usage, it seems sensible to me that we could move the bypass check outside the lock and provide a comment explicitly.

Reply:
I've gone with this solution which has allowed for a non-blocking normal case solution. This also allowed to roll back some of the other acquire and deprecated changes, etc. See latest!

Reply:
Just chiming in to say that the changes are taking shape nicely, let us know when you're done noodling 😸

Reply:
I think this is ready for another review.

🙏 thank you for patience.